posthog/hogql/ast.py

```diff
@@ -309,6 +309,19 @@ class OrderExpr(Expr):
     order: Literal["ASC", "DESC"] = "ASC"
 
 
+class ArrayAccess(Expr):
+    left: Expr
+    right: Expr
```

Comment:
Why `left` and `right` instead of something more concrete like `array` and `subscript`?